core/chaincode/handler.go

```
@@ -662,6 +666,53 @@ func (h *Handler) HandleGetState(msg *pb.ChaincodeMessage, txContext *Transactio
 	return &pb.ChaincodeMessage{Type: pb.ChaincodeMessage_RESPONSE, Payload: res, Txid: msg.Txid, ChannelId: msg.ChannelId}, nil
 }
 
+// Handles query to ledger to get state
+func (h *Handler) HandleGetStateBatch(msg *pb.ChaincodeMessage, txContext *TransactionContext) (*pb.ChaincodeMessage, error) {
+	getStateBatch := &pb.GetStateBatch{}
+	err := proto.Unmarshal(msg.Payload, getStateBatch)
+	if err != nil {
+		return nil, errors.Wrap(err, "unmarshal failed")
+	}
+
+	var stateValues [][]byte
+	chaincodeName := h.ChaincodeName()
+	collection := getStateBatch.Collection
+	chaincodeLogger.Debugf("[%s] getting batch state for chaincode %s, channel %s", shorttxid(msg.Txid), chaincodeName, txContext.ChainID)
+
+	if isCollectionSet(collection) {
+		if txContext.IsInitTransaction {
+			return nil, errors.New("private data APIs are not allowed in chaincode Init()")
+		}
+		if err := errorIfCreatorHasNoReadAccess(chaincodeName, collection, txContext); err != nil {
+			return nil, err
+		}
+		stateValues, err = txContext.TXSimulator.GetPrivateDataMultipleKeys(chaincodeName, collection, getStateBatch.Keys)
+
+	} else {
+		stateValues, err = txContext.TXSimulator.GetStateMultipleKeys(chaincodeName, getStateBatch.Keys)
+	}
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	if stateValues == nil {
+		chaincodeLogger.Debugf("[%s] No state associated with keys. Sending %s with an empty payload", shorttxid(msg.Txid), pb.ChaincodeMessage_RESPONSE)
+	}
+
+	// Constructing response payload
+	protoKvs := make([]*pb.StateKV, 0)
+	for i, key := range getStateBatch.Keys {
+		protoKvs = append(protoKvs, &pb.StateKV{
+			Key:   key,
+			Value: stateValues[i],
+		})
+	}
+
+	payloadBytes, _ := proto.Marshal(&pb.QueryKVs{Kvs: protoKvs})
```

did you intentionally ignore the error?